cpp/arcticdb/pipeline/frame_slice.hpp

```diff
@@ -214,6 +220,7 @@ struct RangesAndKey {
     RowRange row_range_;
     ColRange col_range_;
     entity::AtomKey key_;
+    bool is_incomplete_{false};
```

Comment:
It feels a bit dirty to add this here, but since the main point of this struct is an input parameter to reads (to which we may or may not need to chain the extra validation) it actually seems fairly suitable.

Reply:
Yeah I'm fine with it. The implication is of not incomplete is that it's been indexed before somewhere, so that even in the case where we were merging timeseries together it is at least correct wherever it comes from (and so any schema mismatch errors would ideally happen at a higher level). Whereas if it's incomplete then all bets are off, it's just some isolated data that a user wrote and might contain anything